python/docs/source/user_guide/pandas_on_spark/supported_status_rst_generator.py

```diff
@@ -0,0 +1,225 @@
+from enum import Enum, unique
+from inspect import getmembers, isclass, isfunction, signature
+from typing import Callable, Dict, List, Set, TextIO
+
+import pandas as pd
+import pandas.core.groupby as pdg
+import pandas.core.window as pdw
+import pyspark.pandas as ps
+import pyspark.pandas.groupby as psg
+import pyspark.pandas.window as psw
+
+MAX_MISSING_PARAMS_SIZE = 5
+COMMON_PARAMETER_SET = {"kwargs", "args", "cls"}
+FILE_PATH_PREFIX = "./user_guide/pandas_on_spark"
+HEADER_DOC_FILE = f"{FILE_PATH_PREFIX}/supported_pandas_api_header.txt"
+TARGET_DOC_FILE = f"{FILE_PATH_PREFIX}/supported_pandas_api.rst"
+MODULE_GROUP_MATCH = [(pd, ps), (pdw, psw), (pdg, psg)]
+
+
+@unique
+class Implemented(Enum):
+    IMPLEMENTED = "Y"
+    NOT_IMPLEMENTED = "N"
+    PARTIALLY_IMPLEMENTED = "P"
+
+
+class SupportedStatus:
+    def __init__(self, implemented: str, missing: str = ""):
+        self.implemented = implemented
+        self.missing = missing
+
+
+class SuppportedStatusRSTGenerator:
+    def __init__(self):
+        self.all_supported_status = {}
+
+    def execute(self) -> None:
```

Comment:
Maybe we should better add some docstring with some doctests. That can be enabled by adding the new module into https://github.com/apache/spark/blob/master/dev/sparktestsupport/modules.py#L580